lib/Basic/Version.cpp

```diff
@@ -319,14 +319,16 @@ Optional<Version> Version::getEffectiveLanguageVersion() const {
   switch (Components[0]) {
   case 3:
 #ifdef SWIFT_VERSION_PATCHLEVEL
-    return Version{3, 2, SWIFT_VERSION_PATCHLEVEL};
+    return Version{3, 3, SWIFT_VERSION_PATCHLEVEL};
```

Comment:
Good point. Since that part means means rethinking our design of #if swift, I was planning on doing that separately.  If folks need to have separate code between 4 and 4.1, they can conditionalize those separately as a workaround.  It could lead to some duplication of code and that's why a rethink of #if swift or an introduction of another mechanism might be desirable.